frame/contracts/rpc/src/lib.rs

```diff
@@ -105,17 +103,13 @@ pub enum RpcContractExecResult {
 
 impl From<ContractExecResult> for RpcContractExecResult {
 	fn from(r: ContractExecResult) -> Self {
-		match r {
-			ContractExecResult::Success {
-				flags,
-				data,
-				gas_consumed
-			} => RpcContractExecResult::Success {
-				flags,
-				data: data.into(),
-				gas_consumed,
+		match r.exec_result {
+			Ok(val) => RpcContractExecResult::Success {
+				flags: val.flags.bits(),
+				data: val.data.into(),
+				gas_consumed: r.gas_consumed,
```

Comment:
We should also refactor `RpcContractExecResult` to mirror the new `ContractExecResult` structure, since `gas_consumed` is now available even if there is an error.

Reply:
The thing is that changing the data type of RPCs is a massive PITA because clients (polkadot JS) cannot really detect that it has changed.

This is why I didn't touch the RPC type. I have a follow up in the pipeline which will change this type in a slight way by adding a new type to the now empty `Error` variant. This shouldn't break the clients.

However, if your proposed change is useful to clients we should consider adding a new rpc with the changed type and deprecating this one (in the follow up PR).

Do you think we need the `gas_consumed` on error? @jacogr

Reply:
I talked to @jacogr . He thinks we should absolutely add this value to the RPC but renaming the RPC is not worth it for that because the name is so nice. So we will just break it and let him figure out the rest.

However, not in this PR because I already have another one lined up which will touch the RPC and it is better to do this all in one go to not have more versions than necessary floating around.